site/content/posts/2021-06-21-windows-e2e.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Running E2E Tests On Windows Clusters
+image: /img/sonobuoy.svg
+excerpt: Sonobuoy has new plugins which make running the Kubernetes E2E tests on Windows clusters much more simple.
+author_name: John Schnake
+author_url: https://github.com/johnschnake
+categories: [kubernetes, sonobuoy, windows]
+# Tag should match author to drive author pages
+tags: ['Sonobuoy Team', 'John Schnake']
+date: 2021-06-21
+slug: windows-e2e-tests-with-sonobuoy
+---
+
+In the previous [blog post][lastBlog], we introduced Sonobuoy support for Windows, showed how to set up a Kubernetes cluster which includes Windows nodes, and ran a custom Windows plugin.
+
+In this post, I would like to focus on the running the "E2E tests, the Kubernetes end-to-end tests, against your cluster. Instead of heavily modifying the e2e plugin itself in order to run on Windows, we've provided a new plugin (3 variants of a plugin really) which wrap all the necessary configuration together. This means that, ultimately, you just have to run
+
+```
+sonobuoy run -p NewPlugin.yaml
+```
+
+But if you are curious what these plugins are doing differently and how Windows E2E tests are different upstream, keep reading. By the end of this post you should be able to run the e2e tests against a cluster which includes Windows node and understand how it is different from running them against a Linux-only cluster.
+
+*Note: For most of this blog we will discuss the logic of the commands/configuration but you will not need to run commands directly to follow along. This is because, when all is said and done, Sonobuoy has automated all of these steps.*
+
+## Getting A Windows Cluster
+
+If you would like to run Sonobuoy from a Windows node, you need to set up a Kubernetes cluster with Windows nodes. One of the easiest ways to get started is to use an Azure account and create an AKS cluster. It is easy but does require an account and costs based on resources used. Find more instructions [here](https://docs.microsoft.com/en-us/azure/aks/windows-container-cli).
+
+For the purposes of this blog post, you can start with a 2 node cluster:
+- 1 control plane, Linux node 
+- 1 Windows node for workloads
+
+We will discuss using a cluster with Linux workload clusters later in this post.
```

Comment:
Word `clusters` should be removed

Reply:
oops; yea that should read `workload nodes`

Reply:
All clusters will have linux nodes though, right? I think @vladimirvivien was probably on the money here and have something like 

```suggestion
We will discuss using a cluster with Linux workloads later in this post.
```

Reply:
All clusters have at least one linux nodes in the form of the control plane; not every cluster has to have linux nodes that are not part of the control plane though.

👍 I see why you are saying that can be confusing to just refer to 'nodes' though.